lib/SILGen/SILGenPattern.cpp

```diff
@@ -1031,24 +1031,23 @@ void PatternMatchEmission::emitWildcardDispatch(ClauseMatrix &clauses,
   bool hasGuard = guardExpr != nullptr;
   assert(!hasGuard || !clauses[row].isIrrefutable());
 
-  auto stmt = clauses[row].getClientData<CaseStmt>();
+  auto stmt = clauses[row].getClientData<Stmt>();
   bool hasMultipleItems = false;
-  if (stmt->getKind() == StmtKind::Case) {
-    ArrayRef<CaseLabelItem> labelItems = stmt->getCaseLabelItems();
-    hasMultipleItems = labelItems.size() > 1;
+  if (isa<CaseStmt>(stmt)) {
+    hasMultipleItems = dyn_cast<CaseStmt>(stmt)->getCaseLabelItems().size() > 1;
   }
-  
+
   // Bind the rest of the patterns.
   bindIrrefutablePatterns(clauses[row], args, !hasGuard, hasMultipleItems);
 
   // Emit the guard branch, if it exists.
   if (guardExpr) {
-    if (stmt->getKind() == StmtKind::Case) {
-      SGF.usingImplicitVariablesForPattern(clauses[row].getCasePattern(), stmt, [&]{
+    if (isa<CaseStmt>(stmt)) {
+      SGF.usingImplicitVariablesForPattern(clauses[row].getCasePattern(), dyn_cast<CaseStmt>(stmt), [&]{
```

Comment:
dyn_cast returns nullptr on failure -- I think on this line you want to use cast<> to get an assert here, instead of a null pointer dereference further down